opensrp-giz-malawi/src/main/java/org/smartregister/giz/view/NavigationMenu.java

```diff
@@ -0,0 +1,333 @@
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.github.ybq.android.spinkit.style.FadingCircle;
+
+import org.apache.commons.lang3.StringUtils;
+import org.smartregister.domain.FetchStatus;
+import org.smartregister.giz.R;
+import org.smartregister.giz.adapter.NavigationAdapter;
+import org.smartregister.giz.application.GizMalawiApplication;
+import org.smartregister.giz.contract.NavigationContract;
+import org.smartregister.giz.model.NavigationOption;
+import org.smartregister.giz.presenter.NavigationPresenter;
+import org.smartregister.giz.util.GizConstants;
+import org.smartregister.p2p.activity.P2pModeSelectActivity;
+import org.smartregister.receiver.SyncStatusBroadcastReceiver;
+import org.smartregister.util.PermissionUtils;
+
+import java.lang.ref.WeakReference;
+import java.text.MessageFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+public class NavigationMenu implements NavigationContract.View, SyncStatusBroadcastReceiver.SyncStatusListener {
+    private static NavigationMenu instance;
+    private static WeakReference<Activity> activityWeakReference;
+    private String TAG = NavigationMenu.class.getCanonicalName();
+    private DrawerLayout drawer;
+    private Toolbar toolbar;
+    private NavigationAdapter navigationAdapter;
+    private RecyclerView recyclerView;
+    private TextView tvLogout;
+    private View rootView = null;
+    private ImageView ivSync;
+    private ProgressBar syncProgressBar;
+    private NavigationContract.Presenter mPresenter;
+    private View parentView;
+    private List<NavigationOption> navigationOptions = new ArrayList<>();
+
+    private NavigationMenu() {
+
+    }
+
+    public static NavigationMenu getInstance(Activity activity, View parentView, Toolbar myToolbar) {
+        SyncStatusBroadcastReceiver.getInstance().removeSyncStatusListener(instance);
+        activityWeakReference = new WeakReference<>(activity);
+        int orientation = activity.getResources().getConfiguration().orientation;
+        if (orientation == Configuration.ORIENTATION_PORTRAIT) {
+            if (instance == null) {
+                instance = new NavigationMenu();
+            }
+            SyncStatusBroadcastReceiver.getInstance().addSyncStatusListener(instance);
+            instance.init(activity, parentView, myToolbar);
+            return instance;
+        } else {
+            return null;
+        }
+    }
+
+    private void init(Activity activity, View myParentView, Toolbar myToolbar) {
+        // parentActivity = activity;
+        try {
+            setParentView(activity, parentView);
+            toolbar = myToolbar;
+            parentView = myParentView;
+            mPresenter = new NavigationPresenter(this);
+            registerDrawer(activity);
+            prepareViews(activity);
+        } catch (Exception e) {
+            Log.e(TAG, e.toString());
```

Comment:
Log using Timber and the stack-trace for easy trace-ability

Reply:
Fixed